runtimes/runtimes/lsp/router/util.ts

```diff
@@ -51,3 +51,13 @@ export function mergeObjects(obj1: any, obj2: any) {
     }
     return merged
 }
+
+export function findDuplicates<T>(array: T[]): T[] | undefined {
+    const seen = new Set<T>()
+    const dups = array
+        .filter(a => a !== undefined)
```

Comment:
Non `undefined` values (empty strings, any other non-string falsy values) can pass this filter and result in error in calling code.

Reply:
Although this function is used for strings now, it's not for strings specifically. Empty strings will also be checked for duplicates, which is fine I think.